src/platform/packages/shared/kbn-lens-embeddable-utils/config_builder/schema/charts/xy.ts

```diff
@@ -0,0 +1,661 @@
+          {
+            collapse_by: schema.maybe(collapseBySchema),
+            color: schema.maybe(colorMappingSchema),
+            aggregate_first: schema.maybe(
+              schema.boolean({
+                meta: { description: 'Whether to aggregate before splitting series' },
+              })
+            ),
+          },
+          { meta: { description: 'Split series configuration with color mapping' } }
+        )
+      )
+    ),
+    y: schema.arrayOf(
+      mergeAllMetricsWithChartDimensionSchemaWithRefBasedOps(
+        schema.object(
+          {
+            axis: schema.maybe(schema.oneOf([schema.literal('left'), schema.literal('right')])),
+            color: schema.maybe(staticColorSchema),
+          },
+          { meta: { description: 'Y-axis metric configuration with axis assignment' } }
+        )
+      ),
+      { meta: { description: 'Array of metrics to display on Y-axis' } }
+    ),
+    x: schema.maybe(
+      mergeAllBucketsWithChartDimensionSchema(
+        schema.object({}, { meta: { description: 'X-axis bucket configuration' } })
+      )
+    ),
+  },
+  { meta: { description: 'Data layer for standard queries with metrics and buckets' } }
+);
+
+/**
+ * Data layer configuration for ES|QL queries with column-based metrics
+ */
+const xyDataLayerSchemaESQL = schema.object(
+  {
+    ...layerSettingsSchemaRaw,
+    ...datasetEsqlTableSchema,
+    ...xyDataLayerSharedSchema,
+    breakdown_by: schema.maybe(esqlColumnSchema),
+    y: schema.arrayOf(
+      esqlColumnSchema.extends(
+        {
+          axis: schema.maybe(schema.oneOf([schema.literal('left'), schema.literal('right')])),
+          color: schema.maybe(staticColorSchema),
+        },
+        { meta: { description: 'ES|QL column for Y-axis metric' } }
+      ),
+      { meta: { description: 'Array of ES|QL columns for Y-axis metrics' } }
+    ),
+    x: schema.maybe(esqlColumnSchema),
+  },
+  { meta: { description: 'Data layer for ES|QL queries with column references' } }
+);
+
+/**
+ * Text label configuration for reference lines
+ */
+const referenceLineTextLabelOption = schema.object(
+  {
+    type: schema.literal('label'),
+    text: schema.string({ meta: { description: 'Label text to display' } }),
+  },
+  { meta: { description: 'Text label for reference line' } }
+);
+
+/**
+ * Common visual configuration for reference lines
+ */
+const referenceLineLayerShared = {
+  fill: schema.maybe(
+    schema.oneOf([schema.literal('above'), schema.literal('below')], {
+      meta: { description: 'Fill direction for reference line' },
+    })
+  ),
+  text: schema.maybe(referenceLineTextLabelOption),
+  icon: schema.maybe(
+    schema.string({ meta: { description: 'Icon to display on the reference line' } })
+  ),
+  stroke_width: schema.maybe(schema.number({ meta: { description: 'Line width in pixels' } })),
+  stroke_dash: schema.maybe(
+    schema.oneOf([schema.literal('straight'), schema.literal('dashed'), schema.literal('dotted')], {
```

Comment:
We can probably follow the CSS border-style terminology here: `solid`, `dashed`, `dotted`